src/backend/gl/src/window/egl.rs

```diff
@@ -285,10 +285,27 @@ impl hal::Instance<crate::Backend> for Instance {
 
     fn enumerate_adapters(&self) -> Vec<hal::adapter::Adapter<crate::Backend>> {
         let inner = self.inner.lock();
-        inner
-            .egl
-            .make_current(inner.display, None, None, Some(inner.context))
-            .unwrap();
+        if inner.supports_surfaceless_context {
+            inner
+                .egl
+                .make_current(inner.display, None, None, Some(inner.context))
+                .unwrap();
+        } else {
+            let attributes = [egl::WIDTH, 1, egl::HEIGHT, 1, egl::NONE];
+            let surface = inner
+                .egl
+                .create_pbuffer_surface(inner.display, inner.config, &attributes)
```

Comment:
Why do we need a PBuffer surface? We request the surface-less context only because we can't use the provided surfaces efficiently. If the context has to have them, it's ok, we shouldn't need to do anything, still

Reply:
You mean `create_window_surface` and destroy after `make_current` ?

Reply:
Sorry, I don't understand. Let me try again.

So this is creating a PBuffer because otherwise we can't call `make_current`, correct?
In this case, could we perhaps make it a 1x1 buffer instead of having the full width/height?

Also, we need to keep it around, so that we can call `make_current` later again (see the other comment we are discussing)

Reply:
> So this is creating a PBuffer because otherwise we can't call make_current, correct?

Correct.

> In this case, could we perhaps make it a 1x1 buffer instead of having the full width/height?

```rust
let attributes = [egl::WIDTH, 1, egl::HEIGHT, 1, egl::NONE];
```

It should be 1x1.

Reply:
Sorry, I just missed it. The code was good!